mobile/src/bundles/common/components/input/input.tsx

```diff
@@ -7,42 +7,68 @@ import {
 import { type TextInputProps } from 'react-native';
 import { TextInput } from 'react-native';
 
+import { Text, View } from '~/bundles/common/components/components';
+import { TextCategory } from '~/bundles/common/enums/enums';
 import { useFormController } from '~/bundles/common/hooks/hooks';
-import { globalStyles } from '~/bundles/common/styles/global-styles';
+import { globalStyles } from '~/bundles/common/styles/styles';
 
 import { styles } from './styles';
 
 type Properties<T extends FieldValues> = TextInputProps & {
     control: Control<T, null>;
     name: FieldPath<T>;
     hasError?: boolean;
+    marker?: string;
 };
 
 const Input = <T extends FieldValues>({
     editable = true,
     hasError,
     control,
     name,
+    marker,
+    multiline = false,
     ...props
 }: Properties<T>): JSX.Element => {
     const { field } = useFormController({ name, control });
     const { value, onChange, onBlur } = field;
 
     return (
-        <TextInput
-            onChangeText={onChange}
-            value={value}
-            onBlur={onBlur}
-            {...props}
+        <View
             style={[
-                globalStyles.pl10,
-                globalStyles.borderRadius5,
-                globalStyles.Input,
-                styles.input,
-                !editable && styles.disabled,
-                hasError && styles.error,
+                globalStyles.flexDirectionRow,
+                globalStyles.justifyContentCenter,
+                globalStyles.alignItemsStretch,
             ]}
-        />
+        >
+            {marker && (
+                <View
+                    style={[
+                        globalStyles.justifyContentCenter,
+                        globalStyles.ph20,
+                        styles.marker,
+                    ]}
+                >
+                    <Text category={TextCategory.INPUT}>{marker}</Text>
+                </View>
+            )}
+            <TextInput
+                onChangeText={onChange}
+                value={value.toString()}
```

Comment:
it's not a string by default?

Reply:
Yes, it's a legacy from when I didn't pass my own keyboardType. I don't need it now.